DSCResources/cChocoPackageInstall/cChocoPackageInstall.psm1

```diff
@@ -262,17 +269,22 @@ function UninstallPackage
 
     $env:Path = [Environment]::GetEnvironmentVariable('Path','Machine')
 
-    #Todo: Refactor
-    if (-not ($pParams))
-    {
-        Write-Verbose -Message 'Uninstalling Package Standard'
-        $packageUninstallOuput = choco uninstall $pName -y
+    [string]$chocoParams = "-y"
+    if ($pParams) {
+        $chocoParams += " --params=`"$pParams`""
     }
-    elseif ($pParams)
-    {
-        Write-Verbose -Message "Uninstalling Package with params $pParams"
-        $packageUninstallOuput = choco uninstall $pName --params="$pParams" -y
+    if ($pVersion) {
+        $chocoParams += " --version=`"$pVersion`""
     }
+    # Check if Chocolatey version is Greater than 0.10.4, and add --no-progress 
+    $chocoVer = [system.version](Get-ChocoInstalledPackage | Where-Object { $_.name -eq 'chocolatey' }).Version
+    if ($chocoVer -ge '0.10.4') {

```

Comment:
For explicit sake, I'd like this to be `if ($chocoVer -ge ([System.Version]'0.10.4')) {` ~~, otherwise you may be comparing two string values. It looks correct and works most of the time, but then comes very hard to debug issues and everything looks correct~~.

Actually, it looks like it converts it to System.Version to make the check instead of converting `$chocoVer` to String format for the operation.

~~~
[System.Version]$chocoVersion = '0.10.010'
$chocoVersion.ToString() -ge '0.10.09'
$chocoVersion.ToString() -ge '0.10.011'
$chocoVersion -ge ([System.Version]'0.10.011')
~~~

Good on PowerShell there. It may not be a bad idea to be explicit anyway in case that changes or is different in any future version of PowerShell.

Reply:
I had verified that it was comparing [System.Version] types but I would agree being explicit is better than relying on magic.